subproviders/filters.js

```diff
@@ -376,15 +384,22 @@ LogFilter.prototype.validateLog = function(log){
   return topicsMatch
 }
 
-LogFilter.prototype.update = function(log){
+LogFilter.prototype.update = function(logs){
   // console.log('LogFilter - update')
   const self = this
   // validate filter match
-  var validated = self.validateLog(log)
-  if (!validated) return
-  // add to results
-  self.updates.push(log)
-  self.allResults.push(log)
+  var validLogs = []
+  logs.forEach(function(log) {
+    var validated = self.validateLog(log)
+    if (!validated) return
+    // add to results
+    validLogs.push(log)
+    self.updates.push(log)
+    self.allResults.push(log)
+  })
+  if (validLogs.length > 0) {
+    self.emit('data', validLogs)
```

Comment:
does this also need the null error event?

Reply:
no, the only consumer of the event is the `SubscriptionSubprovider` - would rather emit `error` in that case.